src/django/pfb_analysis/models.py

```diff
@@ -43,8 +43,11 @@ def __repr__(self):
 
     def save(self, *args, **kwargs):
         """ Override to do validation checks before saving, which disallows blank state_abbrev """
+        if not self.label:
+            self.label = slugify(self.name)
```

Comment:
`slugify` isn't imported anywhere. 

Since `self.name` is a slugfield, its probably best to make it read-only and just set it on every save by doing `self.name = slugify(self.label)`.